### PR TITLE
get_predicted: `type` vs. `predict` disambiguation

### PR DESCRIPTION
**Pull request by vincentarelbundock**
This is a PR to address issue #431  

I included a bunch of new tests, and what feels like "conservative" warnings and errors.

From my perspective, the main thing that still needs to be hashed out is what to do with the current `predict="response"` argument value. In Issue #432, there are some suggestions for different naming convention. At this stage, I would actually advocate removing it altogether for the reasons stated in the other issue. 
 
Any feedback would be much appreciated! I'm happy to make any change you deem necessary.

### CODE REVIEW

**Comment by bwiernik** [comment]
I think let's go with "classification".

**Comment by vincentarelbundock** [comment]
Thanks! For what it's worth, I also think "classification" is the best of the available options.

I force-pushed an update with the name change.

**Review by DominiqueMakowski — APPROVED** [review]
thanks, looks good to me :)

**Comment by vincentarelbundock** [comment]
> thanks, looks good to me :)

And right as you said that, I noticed that the class-specific `get_predicted` methods had not all been updated. Sorry! Pushing the fix in 2 minutes.

File: R/get_predicted.R [hunk 1]
```diff
@@ -6,7 +6,14 @@
...
+#' @param predict string or `NULL`
+#' * `"link"` returns predictions on the model's link-scale (for logistic models, that means the log-odds scale) with a confidence interval (CI).
+#' * `"expectation"` (default) also returns confidence intervals, but this time the output is on the response scale (for logistic models, that means probabilities).
+#' * `"prediction"` also gives an output on the response scale, but this time associated with a prediction interval (PI), which is larger than a confidence interval (though it mostly make sense for linear models).
+#' * `"classification"` only differs from `"predict"` for binomial models where it additionally transforms the predictions into the original response's type (for instance, to a factor).
```
Comment (strengejacke) [inline R/get_predicted.R:13]: I think it should be `"prediction"` here-

File: R/get_predicted.R [hunk 2]
```diff
@@ -6,7 +6,14 @@
...
+#' * Other strings are passed directly to the `type` argument of the `predict` method supplied by the modelling package.
+#' * When `predict = NULL`, alternative arguments such as `type` will be captured by the `...` ellipsis and passed directly to the `predict` method supplied by the modelling package.
```
Comment (strengejacke) [inline R/get_predicted.R:15]: I prefer adding parens when referring to functions/methods, so you may change to `predict()` here.

File: R/get_predicted.R [hunk 3]
```diff
@@ -634,25 +613,58 @@ get_predicted.faMain <- function(x, data = NULL, ...) {
   # CI
   if (is.null(ci)) ci <- 0
 
+  # check `predict` user-input
+  supported <- c("expectation", "link", "prediction", "classification")
+  if (isTRUE(verbose) && !is.null(predict) && !predict %in% supported) {
+    warning(sprintf('"%s" is not officially supported by the `get_predicted` function as a value for the `predict` argument. It will not be processed or validated, and will be passed directly to the `predict` method supplied by the modeling package. Users are encouraged to check the validity and scale of the results. Set `verbose=FALSE` to silence this warning, or use one of the supported values for the `predict` argument: %s.', predict, paste(sprintf('"%s"', supported), collapse = ", ")))
```
Comment (strengejacke) [inline R/get_predicted.R:619]: Please wrap in `format_message()`.

File: R/get_predicted.R [hunk 4]
```diff
@@ -634,25 +613,58 @@ get_predicted.faMain <- function(x, data = NULL, ...) {
...
+  if (is.null(dots$type)) {
+    predict_arg <- predict
+    if (is.null(predict)) {
+      stop("Please supply a value for the `predict` argument.")
+    }
+  } else {
+    if(is.null(predict)) {
+      predict_arg <- dots$type
+    } else {
+      stop('The `predict` and `type` arguments cannot be used simultaneously. The preferred argument for the `get_predicted` function is `predict`. If you need to pass a `type` argument directly to the `predict` method associated with your model type, you must set `predict` to `NULL` explicitly: `get_predicted(model, predict=NULL, type="response")`')
```
Comment (strengejacke) [inline R/get_predicted.R:636]: Please wrap in `format_message()`.

**Review by strengejacke — CHANGES_REQUESTED** [review]

**Comment by strengejacke** [comment]
In general, should we have `expected` as alias for `expectation` (and likewise, `predicted` for `prediction`)? At least for me, these shorter forms are more intuitive.

**Comment by vincentarelbundock** [comment]
I'm not a big fan of aliases like this. I think they just introduce complexity to save a couple letters (and all IDEs do completion). I don't have a strong preference for one version or the other (beyond concern for backward compatibility), but I do think we should stick to one canonical way.

**Comment by strengejacke** [comment]
> and all IDEs do completion

Yes, but not for argument-values inside quotes, I think. But I see your point, having too many options might be confusing (or maybe we have them as undocumented options?).

**Comment by bwiernik** [comment]
I agree that we probably shouldn't document too many options/aliases. But I would be okay with silently accepting `expected` and `predicted` as aliases for expectation/prediction

**Comment by vincentarelbundock** [comment]
I pushed a commit with the 4 changes that @strengejacke requested. I think the the Issues I raised and proposed addressing myself should be done in a separate PR. From my perspective this is good to go, but I'm still happy to make any change you want.

FWIW, I also think we should minimize the number of undocumented aliases. It feels safer to be strict about user and to raise informative warnings and errors, rather than do automagical and undocumented changes to user input.

In the current PR, if users call `get_predicted(predict = "expected")`, they will get a warning about unknown argument value.

**Comment by vincentarelbundock** [comment]
To be clear, I don't hold the views expressed above very strongly. None of this feels like a very big deal to me.

**Comment by DominiqueMakowski** [comment]
> But I would be okay with silently accepting expected and predicted as aliases for expectation/prediction

Same

**Comment by strengejacke** [comment]
Thanks for this PR!

**Pull request merged**